### PR TITLE
Extended autocreate test and fixed small bug in autocreate feature

### PR DESCRIPTION
**Pull request by hAdamm**

### CODE REVIEW

File: tests/TestCases/Entity/AutoCreateTest.phpt [hunk 1]
```diff
@@ -13,71 +13,148 @@ class AutoCreateTest extends \Tester\TestCase {
...
+    public function testWriteAutocreateRelationshipOnNonpersistentOwner() {
+        $contact = $this->createNonpersistentContact();
+
         $contact->setValuesPropertiesFromArray(["administrativeEmail_value" => static::DEFAULT_CONTACT_EMAIL]);
         \Tester\Assert::true($contact->administrativeEmail instanceof \obo\Tests\Assets\Entities\Contacts\AdditionalInformation\Email);
         \Tester\Assert::same(self::DEFAULT_CONTACT_EMAIL, $contact->administrativeEmail->value);
     }
 
-    public function testWriteOnPersistentOwner() {
+    public function testReadAutocreateRelationshipOnPersistentOwner() {
```
Comment (ZZromanZZ) [inline tests/TestCases/Entity/AutoCreateTest.phpt]: autoCreate

Comment (ZZromanZZ) [inline tests/TestCases/Entity/AutoCreateTest.phpt]: autoCreate with capital C, in multiple method definitions

File: tests/TestCases/Entity/AutoCreateTest.phpt [hunk 2]
```diff
@@ -13,71 +13,148 @@ class AutoCreateTest extends \Tester\TestCase {
...
     const DEFAULT_CONTACT_EMAIL = "john.doe@mail.com";
     const DEFAULT_CONTACT_NAME = "John Doe";
+    const DEFAULT_NOTE_TEXT = "default text";
 
-    public function testReadOnNonPersistentOwner() {
-        $contact = $this->createNonPersistentContact();
-        \Tester\Assert::true($contact->administrativeEmail instanceof \obo\Tests\Assets\Entities\Contacts\AdditionalInformation\Email);
-    }
+    public function testReadAutocreateRelationshipOnNonpersistentOwner() {
+        $contact = $this->createNonpersistentContact();
```
Comment (lordjim666) [inline tests/TestCases/Entity/AutoCreateTest.phpt]: proc si dal vsude persistent malym?

Comment (lordjim666) [inline tests/TestCases/Entity/AutoCreateTest.phpt]: sorry, lets all think I wrote it in english :D

File: tests/TestCases/Entity/AutoCreateTest.phpt [hunk 3]
```diff
@@ -13,71 +13,148 @@ class AutoCreateTest extends \Tester\TestCase {
...
+
         $administrativeEmail->setValuesPropertiesFromArray(["contact_name" => static::DEFAULT_CONTACT_NAME]);
         \Tester\Assert::true($administrativeEmail->contact instanceof \obo\Tests\Assets\Entities\Contacts\Contact);
         \Tester\Assert::same(static::DEFAULT_CONTACT_NAME, $administrativeEmail->contact->name);
     }
 
-    public function testWriteOnPersistentInverseOwner() {
+    public function testReadAutocreateInverseRelationshipOnPersistentOwner() {
+        $administrativeEmail = $this->createPersistentAdministrativeEmail();
+
```
Comment (lordjim666) [inline tests/TestCases/Entity/AutoCreateTest.phpt]: ugly space

File: tests/TestCases/Entity/AutoCreateTest.phpt [hunk 4]
```diff
@@ -13,71 +13,148 @@ class AutoCreateTest extends \Tester\TestCase {
...
+        \Tester\Assert::true($administrativeEmail->contact instanceof \obo\Tests\Assets\Entities\Contacts\Contact);
+    }
+
+    public function testWriteAutocreateInverseRelationshipOnPersistentOwner() {
         $administrativeEmail = $this->createPersistentAdministrativeEmail();
+
```
Comment (lordjim666) [inline tests/TestCases/Entity/AutoCreateTest.phpt]: ugly space

File: tests/TestCases/Entity/AutoCreateTest.phpt [hunk 5]
```diff
@@ -13,71 +13,148 @@ class AutoCreateTest extends \Tester\TestCase {
...
+        $contact->setValuesPropertiesFromArray(["note_text" => static::DEFAULT_NOTE_TEXT]);
+        \Tester\Assert::true($contact->note instanceof \obo\Tests\TestCases\Entity\Assets\Entities\Notes\Note);
+        \Tester\Assert::same($contact->entityInformation()->name, $contact->note->ownerEntityName );
+        \Tester\Assert::same(static::DEFAULT_NOTE_TEXT, $contact->note->text);
+    }
+
+    public function testReadAutocreateDynamicInverseRelationshipOnPersistentOwner() {
+        $contact = $this->createPersistentContact();
+
```
Comment (lordjim666) [inline tests/TestCases/Entity/AutoCreateTest.phpt]: why those spaces everywhere?

Comment (hAdamm) [inline tests/TestCases/Entity/AutoCreateTest.phpt]: It's ok. Space separates preparation section and testing section

**Pull request merged**